### ISSUE TITLE
During checkout SR users are unable to tell which Payment Method is default, additional or selected

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13728]
**Describe the bug**
- Screen reader users are unable to tell which Payment Method is default and which one is an additional Payment Method.
- Screen reader users are unable to tell which Payment Method is selected
- Action links are being described as links, because they are semantically links. They should be `<button>`s styled as links, they don't have an `href` to be followed (fixed in #14073).

**Tell us the version of Spartacus**
- Library version: 4.0.0

**To Reproduce**
1. Log in as a registered user
2. Add items to cart and start checkout process.
3. Open any Screen Reader (VO, NVDA, JAWS).
4. Tab to Payment Method cards and listen to screen reader feedback.
5. Hear absence of differentiation between Payment Methods.

**Expected behavior**
- Screen reader users can hear which one is the default Payment Method as soon as they tab into it.
- Screen reader users can hear a differentiator for non-default Payment Method.
- Screen reader users can hear which Payment Method is selected, either the default or an additional Payment Method
- Use as per screenshot below `role="region" aria-label="Default Payment Method" tabindex="0"` and 
`role="region" aria-label="Additional Payment Method" tabindex="0"`
- Change all `<a tabindex="0" class="cx-card-link card-link btn-link cx-action-link">...</a>` to `<button class="cx-card-link cx-action-link"> ... </button>`
- As per screenshot below, use `aria-describedby` and an ID to associate the `<button>` with the "Selected" text state. E.g. `<div class="card-header" id="selectedTxt"> Selected </div>` ... ... `<button aria-describedby="selectedTxt" class="cx-card-link cx-action-link">Use this payment</button>`

**Additional context**
- See screenshots for ticket #13724. Similar issue for Shipping Addresses during checkout. So, similar screen reader vocalization is expected. 

**Screenshots**

![Screen Shot 2021-09-15 at 10 25 47 AM](https://user-images.githubusercontent.com/15113219/133453637-76fbe485-0a9f-4a5a-98ff-f9d4b443740b.png)

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

When in read mode: I can hear which payment is selected (default or additional). 
However, when I'm tabbing through, I can't hear which payment is selected to be used for the checkout. But I can still re-select the payment anyway, so the user can still function. @developpeurweb Can you confirm it's ok plz?

Otherwise, it's ok:

![checkoutPaymentSelected](https://user-images.githubusercontent.com/45409515/148983897-1e8d4126-e16c-4a57-b5c4-5995511df974.png)

![checkoutPaymentSelected1](https://user-images.githubusercontent.com/45409515/148983893-12152601-f404-48e4-9108-a2711d32293e.png)

![checkoutPaymentSelected2](https://user-images.githubusercontent.com/45409515/148983891-aba757da-8b3d-4bba-b604-2f565c2020d7.png)
![checkoutPaymentSelected3](https://user-images.githubusercontent.com/45409515/148983884-d04e8d74-b292-461a-aa0a-2e8b1de1aa51.png)

**Comment by BenjaminEmiliani** [comment]
Hi @KateChuen the intended behaviour is for it to specify if the focused card is 'selected'. I will push the small fix. Thanks for noting this bug

**Comment by KateChuen** [comment]
Retested. When focus is on the "Use payment" button of the selected payment, it says selected now:
![checkoutPaymentDefaultSelected](https://user-images.githubusercontent.com/45409515/149211774-31025afb-05c5-4f3d-a72e-98166610a9b9.png)
